app/controllers/flow/vehicle/finance_handler.rb

```diff
@@ -0,0 +1,25 @@
+module Flow
+  module Vehicle
+    class FinanceHandler
+      class << self
+        def model(session_data)
+          VehicleFinanceForm.new(session_data.slice(*VehicleFinanceForm::VEHICLE_FINANCE_ATTRIBUTES.map(&:to_s)))
+        end
+
+        def form(params)
+          VehicleFinanceForm.new(params.require(:vehicle_finance_form).permit(*VehicleFinanceForm::VEHICLE_FINANCE_ATTRIBUTES))
+        end
+
+        def save_data(cfe_connection, estimate_id, form, session_data)
+          age_form = Vehicle::AgeHandler.model(session_data)
+          date_of_purchase = age_form.vehicle_over_3_years_ago ? 4.years.ago.to_date : 2.years.ago.to_date
+          value_form = Vehicle::ValueHandler.model(session_data)
+          cfe_connection.create_vehicle estimate_id, date_of_purchase:,
+                                                     value: value_form.vehicle_value,
+                                                     loan_amount_outstanding: form.vehicle_finance.presence || 0,
```

Comment:
I've just realised there's an edge case bug here: If a user clicks 'Yes' to 'Are there any payments left on the vehicle?', and fills in an amount, then changes their answer to 'No', then submits, loan_amount_outstanding here will be whatever amount they put in, but it _should_ be zero.

Reply:
Ooh yes thanks for that